src/Chapter3/ExamScore.hs

```diff
@@ -0,0 +1,107 @@
+module Chapter3.ExamScore2 where
+import Data.Char
+import Data.List
+        
+data StudentName = MkStudentName String deriving Show
+data SubjectName = MkSubjectName String deriving Show
+data SubjectMarks = MkSubjectMarks Int deriving Show
+data SubjectArr= MkSubjectArr [SubjectName] deriving Show
+data MarkSheet = MkMarkSheet [(StudentName,[(SubjectName,SubjectMarks)])] deriving Show
+
+subAvg :: MarkSheet -> SubjectArr -> SubjectName -> Float
+subAvg (MkMarkSheet mksheet) (MkSubjectArr subjects) (MkSubjectName subname) = 
```

Comment:
why are you *destructuring* (pattern-matching) on `MarkSheet` to get the wrapped marksheet value (that you are holding in `mksheet`)? Are you really using `mksheet` separately?